hikari/impl/rest_bot.py

```diff
@@ -456,6 +474,7 @@ async def start(
             For more information on the other parameters such as defaults see
             AIOHTTP's documentation.
         """
+        # self._server.start should raise errors.ComponentStateConflictError if it's already running
```

Comment:
This will still spawn check_for_updates task, which we don't want to do if there is a ComponentStateConflictError

Reply:
There isn't any defined behaviour for how far it will have gone or what will have been done before that error is raised and check for updates isn't mutating so i don't see any harm in it as an artifact

Reply:
There is, as check_for_updates will have already been called if the parameter was passed

Reply:
Idk what you're getting at with that by yes it will be called every time you call the function

Reply:
What I mean is, we will always create the task for `check_for_updates`, even if there is a StateConflicError, which may lead to the task falling out of scope and erroring out